nncf/tensorflow/layers/wrapper.py

```diff
@@ -203,10 +206,16 @@ def registry_weight_operation(self, weights_attr, op):
         if weights_attr not in self.weights_attr_ops:
             self.weights_attr_ops[weights_attr] = OrderedDict()
 
-        if op.name in self.weights_attr_ops[weights_attr]:
-            raise RuntimeError('Attempt to apply one operation on layer weight twice')
+        if hasattr(op, 'name'):
```

Comment:
I guess this means that the `registry_weight_operation` should have been type-hinted from the beginning. When I implemented a test model for one of the tests in this PR, it took me some time to figure out that the callable for an `OperationWithWeights` must be neither `tf.keras.layers.Layer`, nor a `FakeQuantize`, but an `NNCFOperation`. BTW, why is it that `FakeQuantize` is not an `NNCFOperation`, but we can still quantize weights in TF?

Reply:
I agree that all methods should have type hints. We are working on it. FakeQuantize is only used to quantize activations. It inherits from the tf.keras.layers.Layer because we want to insert FakeQuantize into the model and use standard TF visualization tools to analyze FQ placement for activations.

Reply:
Added a type hint in this particular case